pkg/cache/v3/simple.go

```diff
@@ -586,10 +587,16 @@ func (cache *snapshotCache) respondDelta(ctx context.Context, snapshot ResourceS
 		systemVersion: snapshot.GetVersion(request.GetTypeUrl()),
 	})
 
-	// Only send a response if there were changes
-	// We want to respond immediately for the first wildcard request in a stream, even if the response is empty
+	// Only send a response if:
+	// 1. there were changes
+	// 2. the first wildcard request in a stream
+	// 3. it's for EDS and there's a warming cluster
+	// in case 2 & 3 is resp is sent even if the response is empty
 	// otherwise, envoy won't complete initialization
-	if len(resp.Resources) > 0 || len(resp.RemovedResources) > 0 || (state.IsWildcard() && state.IsFirst()) {
+	if len(resp.Resources) > 0 ||
```

Comment:
Does this make more sense as a `switch{}` now where we can fall through cases? Having this many `||` seems a bit unreadable

Reply:
updated